src/metabase/integrations/google.clj

```diff
@@ -21,8 +21,16 @@
   (deferred-tru "You'll need an administrator to create a Metabase account before you can use Google to log in."))
 
 (defsetting google-auth-client-id
-  (deferred-tru "Client ID for Google Auth SSO. If this is set, Google Auth is considered to be enabled.")
-  :visibility :public)
+  (deferred-tru "Client ID for Google Sign-In. If this is set, Google Sign-In is considered to be enabled.")
+  :visibility :public
+  :setter (fn [client-id]
+            (if client-id
+              (let [trimmed-client-id (str/trim client-id)]
+                (if-not (str/ends-with? trimmed-client-id ".apps.googleusercontent.com")
+                  (throw (ex-info (tru "Invalid Google Sign-In Client ID: must end with \".apps.googleusercontent.com\"")
+                                  {:status-code 400}))
```

Comment:
This is more of a general observation than something you should feel the need to fix here, but in situations like this where you're checking some condition and exiting the function in some way it's better _not_ to put the non-exiting branch of the code in an `else` clause. You could rewrite this as

```clj
(when-not (str/ends-with? ...)
  (throw ...))
(setting/set-string! ...)
```

instead which lowers the total indentation level of the code which makes everything easier to read and understand. When given the choice, make the code flat.

Btw this applies to any language, not just Clojure.

```c++
if (...) {
  return;
}
DoSomethingElse();
```

is easier to read than 

```c++
if (...) {
  return;
} else {
  DoSomethingElse();
}
```

even tho they both do exactly the same thing

Reply:
It would probably be more readable to write the whole thing like

```clj
(fn [client-id]
  (let [client-id (some-> client-id str/trim)]
    (when (and client-id (str/ends-with? client-id ...))
      (throw ...))
    (setting/set-string! ...))
```

so you avoid all that branching. Just my 2 cents tho. It ultimately doesn't make a big difference

Reply:
That's fair, although I have heard the opposite argument: that it's clearer to put `DoSomethingElse()` inside an `else` block to make it super obvious that it will _only_ execute if the condition is false (i.e. if you're skimming the code and don't notice that the `if` block returns or throws an exception).

In the Clojure case I agree though; will change it here